src/Components/Patient/PatientHome.tsx

```diff
@@ -209,32 +211,39 @@ export const PatientHome = (props: any) => {
     });
   };
 
-  const handlePatientDischarge = (value: boolean) => {
+  const handlePatientDischarge = async (value: boolean) => {
+    setIsSendingDischargeApi(true);
     let dischargeData = Object.assign({}, patientData);
     dischargeData["discharge"] = value;
 
-    Promise.all([
-      // using preDischargeForm form data to update patient data
-      dispatch(
-        patchPatient(formatPreDischargeFormData(preDischargeForm), {
-          id: patientData.id,
-        })
-      ),
-      // discharge call
-      dispatch(dischargePatient({ discharge: value }, { id: patientData.id })),
-    ]).then((response: any) => {
-      if ((response || [])[1]?.status === 200) {
-        let dischargeData = Object.assign({}, patientData);
-        dischargeData["discharge"] = value;
-        setPatientData(dischargeData);
+    // calling patchPatient and dischargePatient together caused problems check https://github.com/coronasafe/care_fe/issues/758
```

Comment:
added this comment if somebody tries to make it concurrent thinking it will be faster?